PastaGA.py

```diff
@@ -1,354 +1,265 @@
-            self.__prob_dict[chrom] = selection_chance
+            self.__probabilities[chrom] = selection_chance
             prev_chance = selection_chance
         pass
 
-    def parent_select(self):
-        #selects parents based on probability dict
-        parentchance_1 = random.random()
-        parentchance_2 = random.random()
-        prob_dic1 = self.__prob_dict
-        prob_dic2 = self.__prob_dict
+    def parentSelect(self):
+        """selects parents based on probability dict
+        """
+        parent_chance_1 = random.random()
+        parent_chance_2 = random.random()
+        prob_dic1 = self.__probabilities
+        prob_dic2 = self.__probabilities
         def_check = 0
         for chrom in prob_dic1:
-            if prob_dic1[chrom] >= parentchance_1:
+            if prob_dic1[chrom] >= parent_chance_1:
                 parent_1 = chrom
                 def_check = 1
             if def_check == 1:
                 break
         def_check = 0
         prev_parent = None
         for chrom in prob_dic2:
-            if prob_dic2[chrom] >= parentchance_2:
+            if prob_dic2[chrom] >= parent_chance_2:
                 parent_2 = chrom
-                #if parent_2 == parent_1:
-                    #if prev_parent != None:
-                        #parent_2 = prev_parent
-                    #if prev_parent == None:
-                        #parent_2 = next(chrom)
+                if parent_2 == parent_1:
+                    if prev_parent == None:
+                        pass
+                    else:
+                        parent_2 = prev_parent
                 def_check = 1
             if def_check == 1:
                 break
             prev_parent = chrom
         parent_tuple = (parent_1, parent_2)
         return parent_tuple
 
-    def dict_producer(self):
-        for chrom in self.__chrom_list:
-            chrom.set_fitvalue()
-            self.__chrom_fitdict[chrom] = chrom.get_fitvalue()
+    def fitnessProducer(self):
+        """creates a dictionary containing pasta types and their fitness value
+        """
+
+        for chrom in self.__chroms:
+            chrom.setFitValue()
+            self.__chrom_fit[chrom] = chrom.fitvalue
         pass
 
-    def breed_et_mutate(self, couple_tuple):
-        #mutates children after bred from two chromosomes
+    def breedEtMutate(self, couple_tuple, mutate_chance):
+        """mutates children after bred from two chromosomes
+        """
         pasta1 = couple_tuple[0]
         pasta2 = couple_tuple[1]
-        ingredients_1 = pasta1.list_ing()
-        ingredients_2 = pasta2.list_ing()
+        ingredients_1 = pasta1.listIng()
+        ingredients_2 = pasta2.listIng()
         ingredients_3 = []
         mutated_child = []
-        for num in range(0,1):
+        for num in range(0, 1):
             random_layout = random.randint(1, 6)
             if random_layout == 1:
```

Comment:
See if you can think of accomplishing the same thing here, without all the explicit calls. One thing to consider, would be  iterating through the list of genes/ingredient, and flip a coin which will have you decide which parent you will select from.